# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ScriptLoadContext.h
# Commit: c11cffaded9c
# Full Hash: c11cffaded9cfed53b04e5c7875278e9a9e55396
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-18 21:52:29
# Regressor Bug: 1742437
# File Overlap Count: 5
# Description:
#   Bug 1742437 - re-scope all generic module/script loader files under js/loader; r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138133
# ==============================================================================

diff -r 0527d227c6b6 -r c11cffaded9c dom/script/ScriptLoadContext.h
--- a/dom/script/ScriptLoadContext.h	Fri Feb 18 12:45:52 2022 +0000
+++ b/dom/script/ScriptLoadContext.h	Fri Feb 18 12:45:53 2022 +0000
@@ -11,6 +11,8 @@
 #include "js/RootingAPI.h"
 #include "js/SourceText.h"
 #include "js/TypeDecls.h"
+#include "js/loader/ScriptKind.h"
+#include "js/loader/ScriptLoadRequest.h"
 #include "mozilla/Atomics.h"
 #include "mozilla/Assertions.h"
 #include "mozilla/CORSMode.h"
@@ -27,19 +29,21 @@
 #include "nsCycleCollectionParticipant.h"
 #include "nsIGlobalObject.h"
 #include "nsIScriptElement.h"
-#include "ScriptKind.h"
+#include "js/loader/ScriptKind.h"
 
 class nsICacheInfoChannel;
 
 namespace JS {
 class OffThreadToken;
+namespace loader {
+class ScriptLoadRequest;
 }
+}  // namespace JS
 
 namespace mozilla {
 namespace dom {
 
 class Element;
-class ScriptLoadRequest;
 
 /*
  * DOM specific ScriptLoadContext.
@@ -87,7 +91,7 @@
   NS_DECL_CYCLE_COLLECTING_ISUPPORTS
   NS_DECL_CYCLE_COLLECTION_SCRIPT_HOLDER_CLASS(ScriptLoadContext)
 
-  void SetRequest(ScriptLoadRequest* aRequest);
+  void SetRequest(JS::loader::ScriptLoadRequest* aRequest);
 
   // PreloaderBase
   static void PrioritizeAsPreload(nsIChannel* aChannel);
@@ -198,7 +202,7 @@
    */
   nsCOMPtr<nsIGlobalObject> mWebExtGlobal;
 
-  RefPtr<ScriptLoadRequest> mRequest;
+  RefPtr<JS::loader::ScriptLoadRequest> mRequest;
 
   // Non-null if there is a document that this request is blocking from loading.
   RefPtr<Document> mLoadBlockedDocument;